### PR TITLE
docs(vital-navigation): Update Navigation docs

### PR DESCRIPTION
**Pull request by misterjones**
## Changes

* Perform standard doc cleanup.
* Change instances of CX/CanvasX to Vital.
* Add "Content Editor Details" where applicable.
* Add screenshots.
* Update doc structure from "CX Navigation" to "Vital Navigation."

## Related Issues

* #1385

### CODE REVIEW

**Review by Guilherme-Almeida-Zeni — APPROVED** [review]

File: packages/vital-navigation/doc/SubMenu.md [hunk 1]
```diff
@@ -95,3 +95,49 @@ const Menu = as(
...
+To change this behavior, and permit Content Editors to add Menu Links to Menu Items with Sub-Menus,
+open
+[`vital-navigation/src/components/Menu/tokens/vitalMenu.ts`](https://github.com/johnsonandjohnson/Bodiless-JS/blob/main/packages/vital-navigation/src/components/Menu/tokens/vitalMenu.ts
+':target=blank'), and comment out (or remove) the following line:
+
+```ts
+Title: flowIf(useHasSubMenu)(as(cxMenuTitle.WithLinkDisabled)),
+```
+
+You should find this line in the `Burger` token, under its `Behavior` property.
```
Comment (hvanyo) [inline packages/vital-navigation/doc/SubMenu.md:143]: @Guilherme-Almeida-Zeni  @misterjones  seems weird this in burger token??

Comment (misterjones) [reply packages/vital-navigation/doc/SubMenu.md:143]: @hvanyo Oh, I just realized that this line is also in `TopNav` token.

See: [`vitalMenu.ts` Line #166](https://github.com/johnsonandjohnson/Bodiless-JS/blob/chore/upgrade-to-gatsby-4-and-tw-3/packages/vital-navigation/src/components/Menu/tokens/vitalMenu.ts#L166)

@Guilherme-Almeida-Zeni, please let me know if one or both of these lines should be commented out, and I will update accordingly.

Comment (Guilherme-Almeida-Zeni) [reply packages/vital-navigation/doc/SubMenu.md:143]: @hvanyo ,

I believe it's in both `TopNav` and `Burger` tokens to disable the menu title link on both menus.

Comment (misterjones) [reply packages/vital-navigation/doc/SubMenu.md]: Thank you, @Guilherme-Almeida-Zeni!

**Updated.** 👍

**Review by hvanyo — APPROVED** [review]

**Pull request merged**